### ISSUE TITLE
can't reproduce best performance result

### ISSUE THREAD

**Issue by KnightHarute** [Henry1iu/TNT-Trajectory-Predition#5]
Hi Liu,

Thanks for sharing the excellent work!
By following your instruction, the VectorNet model and TNT model is successfully been trained.
But evaluation results are not close to the best performance that you reported.

I didn't change any params, just doing:
`python train_vectornet.py --data_root dataset/interm_data --output_dir run/tnt/ \
                          --aux_loss --batch_size 64 --with_cuda --cuda_device 0 \
                          --lr 0.0010 --warmup_epoch 30 --lr_update_freq 10 --lr_decay_rate 0.1`
and 
`python train_tnt.py --data_root dataset/interm_data --output_dir run/tnt/ \
                    --aux_loss --batch_size 64 --with_cuda --cuda_device 0 \
                    --lr 0.0010 --warmup_epoch 30 --lr_update_freq 10 --lr_decay_rate 0.1`

The evaluation for VectorNet is:
`{"minADE": 3.4815248822055596, "minFDE": 7.2300321001234265, "MR": 0.8710478313741387}
{"minADE": 3.111675608063909, "minFDE": 6.603447830912062, "MR": 0.8342369274422375}
{"minADE": 2.9561056829126735, "minFDE": 6.332167925216294, "MR": 0.8328942034860154}
{"minADE": 2.427972885278816, "minFDE": 5.3626797689252, "MR": 0.7882802999594649}
{"minADE": 2.244481276930554, "minFDE": 5.030226468854356, "MR": 0.7672020672882043}
{"minADE": 2.1562082622641476, "minFDE": 4.851252928228176, "MR": 0.7620591811917309}
{"minADE": 2.0795099592553905, "minFDE": 4.642897191516932, "MR": 0.7285164167004459}
{"minADE": 2.0393877537693, "minFDE": 4.533113019578649, "MR": 0.7257042967166599}
{"minADE": 1.987427299962104, "minFDE": 4.481501569552523, "MR": 0.7270723550871504}
{"minADE": 1.9800827187772254, "minFDE": 4.463111162570423, "MR": 0.7233481961897041}
{"minADE": 1.9479401628038777, "minFDE": 4.401293576761737, "MR": 0.7107316578840697}
{"minADE": 1.9075498426525, "minFDE": 4.308581153690744, "MR": 0.6994831779489258}
{"minADE": 1.8781217305841496, "minFDE": 4.245759824918292, "MR": 0.6946189704094041}
{"minADE": 1.864234075624533, "minFDE": 4.197237467402696, "MR": 0.6823824483177949}
{"minADE": 1.857677991022711, "minFDE": 4.239490465223712, "MR": 0.7016872719902716}
{"minADE": 1.847790163231717, "minFDE": 4.193527398336232, "MR": 0.6902867855695176}
{"minADE": 1.843492083015308, "minFDE": 4.178072442196886, "MR": 0.6852959059586542}
{"minADE": 1.8167193511566435, "minFDE": 4.092232324094985, "MR": 0.6764541953790029}
{"minADE": 1.7928084237393596, "minFDE": 4.072192936356568, "MR": 0.670019254154844}
{"minADE": 1.7857650734562092, "minFDE": 4.060924091721804, "MR": 0.6695379002837455}
{"minADE": 1.7815429018976958, "minFDE": 4.045257472212025, "MR": 0.6677391568706932}
{"minADE": 1.7797643037885205, "minFDE": 4.046152594878983, "MR": 0.6667511147142278}
{"minADE": 1.7785541320911624, "minFDE": 4.043385831417258, "MR": 0.6673338062423997}
{"minADE": 1.7779266739869517, "minFDE": 4.041878094977796, "MR": 0.666472436157276}
{"minADE": 1.7767552207522668, "minFDE": 4.037293235213845, "MR": 0.6661177543575193}`

And the evaluation for TNT model is:
`{"minADE": 2.6778897080373825, "minFDE": 4.9689554443311765, "MR": 0.6390605999189298}
{"minADE": 2.3585556863337778, "minFDE": 4.433373124361469, "MR": 0.5934333198216457}
{"minADE": 1.847447761937877, "minFDE": 3.4086198245319834, "MR": 0.48120186461289016}
{"minADE": 1.834315601402311, "minFDE": 3.5277772029447214, "MR": 0.5512008512363195}
{"minADE": 1.7745761108987022, "minFDE": 3.475042892153443, "MR": 0.5198875152006486}
{"minADE": 1.7464681564993911, "minFDE": 3.4607072183849072, "MR": 0.546133968382651}`

**Comment by Henry1iu** [comment]
Hi, 

Thanks for trying my implementation. The given hyperparameters are just examples. You can start to tune your own magic numbers to achieve the best performance.  You can try to enlarge the batch size and change the learning rate. 

As for the best performance achieved, it's still not as good as the performance reported in the paper. I'm still enhancing the model structure. Due to the IP issue, the authors refused to provide more detail to help me. You can definitely try your own idea to approach the best performance reported in the paper. My best performance is just a reference. 

Wish you good luck.

Best regards,
Jianbang

**Comment by Henry1iu** [comment]
Hi, 

I fixed a bug in the trajectory selection. You should be able to reproduce the best-achieved performance using 128 batch size when training the TNT model. 

Best Regards,
Jianbang
